google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/executor/AppEngineExecutor.java

```diff
@@ -30,7 +39,15 @@ public AppEngineExecutor(AppEngineTask task) {
 
   @Override
   public void run() {
-    task.execute(this::setProcess);
+    task.execute(
```

Comment:
just noticed that `ProcessStartListener#onStart` is called _after_ the process has been started. Technically, what we want here is to acquire the read lock _beforeStart_? I don't think practically this makes much of a difference though.

Reply:
what do you mean by "problematic considering current SDK API call sequence"?

Reply:
we _could_ update ProcessStartListener to contain an additional beforeStart method. This clearly would break existing clients though, but not a huge deal to update our 4 plugins. I agree though that it would be very hard to violate this though (and as long as the consequences of such a violation are well presented to the user, I'm ok with it as is and will leave it up to you).